lib/matplotlib/tests/test_legend.py

```diff
@@ -642,3 +642,72 @@ def test_no_warn_big_data_when_loc_specified():
         ax.plot(np.arange(5000), label=idx)
     legend = ax.legend('best')
     fig.draw_artist(legend)  # Check that no warning is emitted.
+
+
+def test_plot_multiple_input_multiple_label():
+    # test ax.plot() with multidimensional input
+    # and multiple labels
+    x = [1, 2, 5]
```

Comment:
As with in the example above I recommend not to use the same number of datasets and data points per dataset. By doing so, you implicitly also test that the dimensions are right; and as above it's easier to understand for the reader of the test.

Same holds for the other tests.

Reply:
Thank you for all the suggestions.
I agree, now tests and examples should be easier to understand.